inbox/mailsync/service.py

```diff
@@ -118,6 +118,25 @@ def run(self):
         while self.keep_running:
             retry_with_logging(self._run_impl, self.log)
 
+        for email_sync_monitor in self.email_sync_monitors.values():
+            if email_sync_monitor.delete_handler:
+                email_sync_monitor.delete_handler.kill()
+            email_sync_monitor.sync_greenlet.kill(block=False)
+            email_sync_monitor.join()
+        self.log.info(
+            "stopped email sync monitors", count=len(self.email_sync_monitors)
+        )
+        for contact_sync_monitor in self.contact_sync_monitors.values():
+            contact_sync_monitor.kill()
+        self.log.info(
+            "stopped contact sync monitors", count=len(self.contact_sync_monitors)
+        )
+        for event_sync_monitor in self.event_sync_monitors.values():
+            event_sync_monitor.kill()
+        self.log.info(
+            "stopped event sync monitors", count=len(self.event_sync_monitors)
+        )
```

Comment:
This is refactored out from the signal handler `stop()` below, with more logging.